src/Core/Path.php

```diff
@@ -29,16 +29,24 @@ public static function join(...$parts)
             $parts = $parts[0];
         }
 
-        // Cleanup and join all parts
-        $parts = array_filter(array_map('trim', array_filter($parts ?? [])));
+        // Remove empty parts and trim all parts
+        // Explicitly allow zero as a segment in the path (e.g. /some/path/0/file.txt)
+        $notEmpty = fn(mixed $part) => ($part === 0 || $part === '0' || (bool) $part);
```

Comment:
```suggestion
        $notEmpty = fn(mixed $part) => (is_string($part) || is_int($part)) && strlen($part) > 0;
```
Or something to this effect

`(bool) $part` look flakey, as lots of non-appropriate things like a non-empty array are true e.g. `var_dump((bool) [1]); => true`

Reply:
The purpose of this PR is not to make `Path` super type-robust or handle any type-flakeyness it already has. The default `array_filter()` callback is effectively just `fn ($value) => (bool) $value;`.

This PR just wants to allow `0` and `'0'` as valid values, and nothing else.